patches/minecraft/net/minecraft/world/gen/ChunkGeneratorEnd.java.patch

```diff
@@ -25,26 +25,35 @@
 +        this.field_73212_b = ctx.getScale();
 +        this.field_185973_o = ctx.getIsland();
 +        this.field_185972_n = (MapGenEndCity) net.minecraftforge.event.terraingen.TerrainGen.getModdedMapGen(this.field_185972_n, net.minecraftforge.event.terraingen.InitMapGenEvent.EventType.END_CITY);
++        this.structureManager = net.minecraftforge.event.terraingen.TerrainGen.getAddedStructures(ChunkGeneratorHell.class);
```

Comment:
1. The arguments seems incorrect.
2. Can you simply leave it in the end of the constructor of the root class?

Reply:
1) Yes
2) Using this.getClass() ?

Reply:
Why not just this?

Reply:
Wasn't sure if it is a good idea to pass a "unfinished" object here. Modders could potentially call methods on this object and cause trouble. Not sure.
Also I couldn't use the generic event anymore.

But if you want I can change this